app/docker/services/imageService.js

```diff
@@ -138,6 +139,32 @@ angular.module('portainer.docker')
     return Image.tag({id: id, tag: imageConfig.tag, repo: imageConfig.repo}).$promise;
   };
 
+  service.downloadImages = function(images) {
+    var deferred = $q.defer();
+    var names = ImageHelper.getImagesNamesForDownload(images);
+    Image.download(names).$promise
```

Comment:
If you do not manipulate the response data, the `deferred` pattern is not required here.

You can simply write:

```javascript
service.downloadImages = function(images) {
  var names = ImageHelper.getImagesNamesForDownload(images);
  return Image.download(names).$promise;
}
```